# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/EditorBase.cpp
# Commit: cdcef4f3f635
# Full Hash: cdcef4f3f63553a28a1055e6f4f051778ca8fb0a
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-09-22 09:56:47
# Regressor Bug: 1789967
# File Overlap Count: 1
# Description:
#   Bug 1789967 - part 2: Make `TextEditor` and `HTMLEditor` implement `EditorBase::CollapseSelectionToEndOfLastLeafNode` by themselves r=m_kato
#   
#   It does different thing for `TextEditor` and `HTMLEditor`, and used almost
#   internally.  Therefore, it should be implemented in the sub classes and
#   we should name them better.
# ==============================================================================

diff -r cc2e8b7c548e -r cdcef4f3f635 editor/libeditor/EditorBase.cpp
--- a/editor/libeditor/EditorBase.cpp	Thu Sep 22 06:06:54 2022 +0000
+++ b/editor/libeditor/EditorBase.cpp	Thu Sep 22 06:17:36 2022 +0000
@@ -5,6 +5,7 @@
 
 #include "EditorBase.h"
 
+#include "ErrorList.h"
 #include "mozilla/DebugOnly.h"  // for DebugOnly
 
 #include <stdio.h>   // for nullptr, stdout
@@ -1343,52 +1344,7 @@
   return rv;
 }
 
-NS_IMETHODIMP EditorBase::EndOfDocument() {
-  AutoEditActionDataSetter editActionData(*this, EditAction::eNotEditing);
-  if (NS_WARN_IF(!editActionData.CanHandle())) {
-    return NS_ERROR_NOT_INITIALIZED;
-  }
-  nsresult rv = CollapseSelectionToEndOfLastLeafNode();
-  NS_WARNING_ASSERTION(
-      NS_SUCCEEDED(rv),
-      "EditorBase::CollapseSelectionToEndOfLastLeafNode() failed");
-  // This is low level API for XUL applcation.  So, we should return raw
-  // error code here.
-  return rv;
-}
-
-nsresult EditorBase::CollapseSelectionToEndOfLastLeafNode() const {
-  MOZ_ASSERT(IsEditActionDataAvailable());
-
-  // XXX Why doesn't this check if the document is alive?
-  if (NS_WARN_IF(!IsInitialized())) {
-    return NS_ERROR_NOT_INITIALIZED;
-  }
-
-  // get the root element
-  Element* rootElement = GetRoot();
-  if (NS_WARN_IF(!rootElement)) {
-    return NS_ERROR_NULL_POINTER;
-  }
-
-  nsIContent* lastLeafContent = rootElement;
-  if (IsTextEditor()) {
-    lastLeafContent = rootElement->GetFirstChild();
-    MOZ_ASSERT(lastLeafContent && lastLeafContent->IsText());
-  } else {
-    for (nsIContent* child = lastLeafContent->GetLastChild();
-         child && HTMLEditUtils::IsContainerNode(*child);
-         child = child->GetLastChild()) {
-      lastLeafContent = child;
-    }
-  }
-
-  nsresult rv =
-      CollapseSelectionToEndOf(OwningNonNull<nsINode>(*lastLeafContent));
-  NS_WARNING_ASSERTION(NS_SUCCEEDED(rv),
-                       "EditorBase::CollapseSelectionToEndOf() failed");
-  return rv;
-}
+NS_IMETHODIMP EditorBase::EndOfDocument() { return NS_ERROR_NOT_IMPLEMENTED; }
 
 NS_IMETHODIMP EditorBase::GetDocumentModified(bool* aOutDocModified) {
   if (NS_WARN_IF(!aOutDocModified)) {